tests/scripts/test_evaluate_release_tag.py

```diff
@@ -0,0 +1,67 @@
+from scripts.evaluate_release_tag import filter_non_alpha_releases, should_build_docs
+import pytest
+from pep440_version_utils import Version
+from typing import List
+from unittest.mock import patch
+
+
+@pytest.mark.parametrize(
+    "releases, expected",
+    [
+        ([Version("1.1.0"), Version("2.2.0")], [Version("1.1.0"), Version("2.2.0")]),
+        (
+            [Version("1.1.0"), Version("2.2.0"), Version("1.1.1a")],
+            [Version("1.1.0"), Version("2.2.0")],
+        ),
+        (
+            [Version("1.1.0"), Version("2.2.0"), Version("1.1.1a1")],
+            [Version("1.1.0"), Version("2.2.0")],
+        ),
+    ],
+)
+def test_filter_non_alpha_releases(releases: List[Version], expected: List[Version]):
+    result = filter_non_alpha_releases(releases)
+    assert result == expected
+
+
+@pytest.mark.parametrize(
+    "releases, tag, expected",
+    [
+        ([Version("1.1.0"), Version("2.2.0")], Version("2.3.0"), True),
+        (
+            [Version("1.1.0"), Version("2.2.0"), Version("2.3.0a1")],
+            Version("2.2.1"),
+            True,
+        ),
+        (
+            [Version("1.1.0"), Version("2.2.0"), Version("2.3.0")],
+            Version("1.2.0"),
+            True,
+        ),
+        (
+            [Version("1.1.0"), Version("1.2.0a1"), Version("2.3.0")],
+            Version("1.1.2"),
+            True,
+        ),
+        (
+            [Version("1.1.0"), Version("2.2.0"), Version("2.3.0")],
+            Version("2.2.1"),
+            False,
```

Comment:
I don't quite understand why we shouldn't build docs if we release a micro for an earlier minor series?

Reply:
Good, point. This is what I have discovered with tests.

Reply:
So the docs should be built only on the latest minor version.